app/topics/show-topic/show-topic.component.ts

```diff
@@ -210,4 +224,12 @@ export class ShowTopicComponent implements OnInit, OnDestroy {
     );
     return this.translatedResponse;
   }
+
+  private showStatusOptionsToCurrentUsers() {
```

Comment:
No need for this function.
Also it can be simplified:
`this.isReadyForReview = this.currentUser.role === 'Student' && this.topic.status === 'InProgress';`